src/espina/client/router.js

```diff
@@ -0,0 +1,101 @@
+/*global window NODE_ENV*/
+import {stringify} from 'query-string';
+import BaseRouter from 'espina/shared/router'
+import Scroll from "scroll-js" 
+
+export default class Router extends BaseRouter {
+
+  get locale(){
+    return Router.locale();
+  }
+
+  initializeHistory(createHistory, store,fnChange) {
+    let router = this;
+    router.history = createHistory();
+    router.history.listen(router.onLocationChange.bind(router, store));
+    if (fnChange) router.history.listen(fnChange);
+  }
+
+  // Changing Route
+
+  // this will cause onLocationChange to fire with
+  // the new location.
+  pushRoute(route_name, action, payload,params,hash){
+    let router = this,
+        route = router.routes.findByName(route_name);
+
+    action = {
+      type: action ? action.getType() : 'UPDATE_LOCATION',
+      payload: payload,
+      no_scroll: payload ? payload.no_scroll : false,
+      transition:true
+    };
+
+    router.pushHistory({
+      pathname: route.url(action , router.i18n, payload,hash),
+      search: stringify(params),
+      state: action,
+      hash: hash,
+    });
+  }
+
+  pushHistory(location){
+
+    this.history.push(location);
+  }
+
+  onLocationChange(store, new_location){
+    let router = this;
+    // NOTE: You can pass true to scrollTop property (ie Integer).
+    router.animateTransition(new_location);
+    let action = new_location.state || {};
+    action.type =  action.type || 'UPDATE_LOCATION'
+    if(!new_location.pathname.startsWith("/"))
+      new_location.pathname = "/"+new_location.pathname;
+    action.location = router.parseLocation(new_location);
+    store.dispatch(action);
+  }
```

Comment:
I'd like to add a configuration option in the routes so that `document.title` can be updated here if the new route  has a page title.